src/modules/blockchainApplication/manage/store/reducer.js

```diff
@@ -22,6 +22,7 @@ const initialState = {
 export const pins = (state = initialState.pins, { type, chainId }) => {
   switch (type) {
     case actionTypes.toggleApplicationPin:
+    case actionTypes.deleteApplicationByChainId:
```

Comment:
The delete logic cannot be placed here, needs a separate case. For example, deleting an application (that is not pinned) will push that application to the pins array.

Reply:
@oskarleonard Nice catch. Fixed